azure-kusto-data/source/managedIdentitiesClient.js

```diff
@@ -1,39 +1,25 @@
-const http = require("http");
-const url = require("url");
-
-module.exports = function acquireToken(resource, msiEndpoint, msiSecret, callback) {
-    const msiResource = `${msiEndpoint}/?resource=${resource}&api-version=2017-09-01`;
-    const msiUrl = url.parse(msiResource);
-
-    const options = {
-        host: msiUrl.host,
-        hostname: msiUrl.hostname,
-        port: msiUrl.port,
-        path: msiUrl.path,
-        headers: {
-            secret: msiSecret
+const request = require("request");
+
+module.exports = function acquireToken(resource, msiEndpoint, msiClientId, callback) {
+    // provided endpoint should be of the following format http://169.254.169.254/metadata/identity/oauth2/token
+    // or http://localhost/oauth2/token (to be deprecated)
+    // for more info refer to: https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview
+    let msiUri = `${msiEndpoint}/?resource=${resource}&api-version=2018-02-01`;
```

Comment:
The `api-version` is different for Azure Functions and App Services. You might want to add is a parameter as well.

Reply:
I think I prefer resolving it. I don't really expect end users to mess around with versions.